pkg/virt-controller/watch/vm.go

```diff
@@ -1285,6 +1294,38 @@ func (c *VMController) updateStatus(vmOrig *virtv1.VirtualMachine, vmi *virtv1.V
 	return err
 }
 
+func (c *VMController) syncReadyConditionFromVMI(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachineInstance) {
+	vmReadyCond := c.getCondition(vm, virtv1.VirtualMachineReady)
+	vmiReadyCond := controller.NewVirtualMachineInstanceConditionManager().
+		GetCondition(vmi, virtv1.VirtualMachineInstanceConditionType(k8score.PodReady))
+
+	if vmiReadyCond != nil {
+		now := v1.NewTime(time.Now())
+		if vmReadyCond == nil {
+			log.Log.Object(vm).V(4).Info("Adding ready condition")
+			newCond := virtv1.VirtualMachineCondition{
+				Type:               virtv1.VirtualMachineReady,
+				Status:             vmiReadyCond.Status,
+				LastProbeTime:      now,
+				LastTransitionTime: now,
+				Reason:             vmiReadyCond.Reason,
+				Message:            vmiReadyCond.Message,
+			}
+			vm.Status.Conditions = append(vm.Status.Conditions, newCond)
+		} else if vmiReadyCond.Status != vmReadyCond.Status {
+			log.Log.Object(vm).V(4).Info("Updating ready condition")
```

Comment:
Why not always update the VM ready condition? regardless of it's value...
If we don't update it, the timestamps (at least the `LastProbeTime` will become stale and it usually indicates of an issue to the user).

Reply:
I wanted to minimize update requests, but this makes sense. Should the `LastProbeTime` be copied from VMI or set to current time?

Reply:
I think that copying `LastProbeTime` makes sense...

Reply:
Done